src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -96,8 +100,12 @@ case class IndexLogEntry(
 
   def numBuckets: Int = derivedDataset.properties.numBuckets
 
-  def plan(spark: SparkSession): LogicalPlan = {
-    LogicalPlanSerDeUtils.deserialize(source.plan.properties.rawPlan, spark)
+  def relation: RelationMetadataEntry = source.relations match {
```

Comment:
This can just return `relations` and the caller side can do the check. (The `def`s here are mostly aliases to provide easy accesses to fields)